signer/tests/integration/complete_deposit.rs

```diff
@@ -0,0 +1,742 @@
+    chain_tip: &BitcoinBlock,
+) -> (CompleteDepositV1, ReqContext) {
+    // Okay now we get ready to create the transaction using the
+    // `CompleteDepositV1` type.
+    let complete_deposit_tx = CompleteDepositV1 {
+        // This outpoint points to the deposit UTXO.
+        outpoint: req.outpoint(),
+        // This amount must not exceed the amount in the deposit request.
+        amount: req.amount,
+        // The recipient must match what was indicated in the deposit
+        // request.
+        recipient: req.recipient.clone().into(),
+        // The deployer must match what is in the signers' context.
+        deployer: StacksAddress::burn_address(false),
+        // The sweep transaction ID must point to a transaction on
+        // the canonical bitcoin blockchain.
+        sweep_txid: sweep_tx.txid.into(),
+        // The block hash of the block that includes the above sweep
+        // transaction. It must be on the canonical bitcoin blockchain.
+        sweep_block_hash: chain_tip.block_hash,
+        // This must be the height of the above block.
+        sweep_block_height: chain_tip.block_height,
+    };
+
+    // This is what the current signer things of the state of things.
+    let ctx = ReqContext {
+        chain_tip: chain_tip.into(),
+        // This value means that the signer will go back 10 blocks when
+        // looking for pending and accepted deposit requests.
+        context_window: 10,
+        // The value here doesn't matter.
+        origin: fake::Faker.fake_with_rng(&mut OsRng),
+        // This value affects how many deposit transactions are consider
+        // accepted.
+        signatures_required: 2,
+        // This is who the current signer thinks deployed the sBTC
+        // contracts.
+        deployer: StacksAddress::burn_address(false),
+    };
+
+    (complete_deposit_tx, ctx)
+}
+
+/// Generate a signer set, deposit requests and store them into the
+/// database.
+async fn deposit_setup<R>(rng: &mut R, pg_store: &PgStore) -> Vec<PublicKey>
+where
+    R: rand::RngCore + rand::CryptoRng,
+{
+    // This is just a sql test, where we use the `TestData` struct to help
+    // populate the database with test data. We set all the other
+    // unnecessary parameters to zero.
+    let num_signers = 7;
+    let test_model_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 20,
+        num_stacks_blocks_per_bitcoin_block: 0,
+        num_deposit_requests_per_block: 2,
+        num_withdraw_requests_per_block: 0,
+        num_signers_per_request: num_signers,
+    };
+
+    // Normal: this generates the blockchain as well as deposit request
+    // transactions in each bitcoin block.
+    let signer_set = testing::wsts::generate_signer_set(rng, num_signers);
+    let test_data = TestData::generate(rng, &signer_set, &test_model_params);
+    test_data.write_to(pg_store).await;
+    signer_set
+}
+
+/// Get the full block
+async fn get_bitcoin_canonical_chain_tip_block(
```

Comment:
You're double-unwrapping this everywhere you use it (so far) -- maybe you should skip the result and double-unwrap here instead to improve readability in tests?

Reply:
Done in https://github.com/stacks-network/sbtc/pull/545/commits/2773454d18c6c8a8776c9485d2e751aefffd6a60